platform/x11/os_x11.cpp

```diff
@@ -1662,6 +1661,7 @@ void OS_X11::handle_key_event(XKeyEvent *p_event, bool p_echo) {
 
 				input->accumulate_input_event(k);
 			}
+			memfree(utf8string);
```

Comment:
I can't say if the rest of the PR is good or not (a review from @hpvb or @RandomShaper would be great!), but I think this one line is definitely wanted so it might be worth making a separate PR with it to get it merged.

Reply:
I moved this to #33555.